test/jdk/java/io/File/TempDirDoesNotExist.java

```diff
@@ -25,91 +25,152 @@
                 }
-            } else {
-                throw new Exception("unknown case: " + arg);
             }
         }
+    }
 
-        if (args.length == 0) {
-            // standard test with default setting for java.io.tmpdir
-            testMessageNotExist(0, ioWarningMsg, "TempDirDoesNotExist", "io");
-            testMessageNotExist(0, ioWarningMsg, "TempDirDoesNotExist", "nio");
-
-            // valid custom java.io.tmpdir
-            testMessageNotExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + userDir,
-                    "TempDirDoesNotExist", "io");
-            testMessageNotExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + userDir,
-                    "TempDirDoesNotExist", "nio");
-
-            // invalid custom java.io.tmpdir
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "io");
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir,
-                    "TempDirDoesNotExist", "nio");
+    private static String tempDir() {
+        String timeStamp = String.valueOf(System.currentTimeMillis());
+        return Path.of(USER_DIR, "non-existing-", timeStamp).toString();
+    }
 
-            // test with security manager
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir
-                            + " -Djava.security.manager",
-                    "TempDirDoesNotExist", "io");
+    public static Stream<Arguments> existingProvider() {
+        return Stream.of(Arguments.of(0, WARNING,
+                                      new String[] {
+                                          "-Djava.io.tmpdir=" +
+                                          tempDir(),
+                                          "TempDirDoesNotExist", "io"
+                                      }),
+                         Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + tempDir(),
+                                "TempDirDoesNotExist", "nio"
+                            }),
+                         Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + tempDir()
+                                + " -Djava.security.manager",
+                                "TempDirDoesNotExist", "io"
+                            }),
+                         Arguments.of(0, WARNING,
+                            new String[] {
+                                "-Djava.io.tmpdir=" + tempDir()
+                                + " -Djava.security.manager",
+                                "TempDirDoesNotExist", "nio"
+                            }));
+    }
 
-            testMessageExist(0, ioWarningMsg, "-Djava.io.tmpdir=" + tempDir
-                            + " -Djava.security.manager",
-                    "TempDirDoesNotExist", "nio");
+    public static Stream<Arguments> nonexistentProvider() {
```

Comment:
The `nonexistentProvider` seems like an odd name. The only provider here is the junit provider (not a FileSystem provider)
How about `NoTempDir` and `TempDir` for the two argument streams.

Reply:
I think maybe instead "tempDirSource' and "noTempDirSource."

Reply:
So changed in 50411b85aeefb2576a149ae126b7d55aad6e2be0.